go/obscuronode/enclave/sql/edgelessdb.go

```diff
@@ -0,0 +1,480 @@
+	if err != nil {
+		return fmt.Errorf("faild to create manifest initialization req - %w", err)
+	}
+	caCertPool := x509.NewCertPool()
+	if ok := caCertPool.AppendCertsFromPEM([]byte(edbPEM)); !ok {
+		return fmt.Errorf("failed to append to CA cert from edb cert pem")
+	}
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs:    caCertPool,
+				MinVersion: tls.VersionTLS12,
+			},
+		},
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		return fmt.Errorf("manifest initialization req failed - %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode < 200 || resp.StatusCode > 299 {
+		var msg []byte
+		_, err := resp.Body.Read(msg)
+		if err != nil {
+			return fmt.Errorf("manifest initialization req failed with status code: %d, failed to read status text", resp.StatusCode)
+		}
+		return fmt.Errorf("manifest initialization req failed with status: %d %s", resp.StatusCode, msg)
+	}
+	return nil
+}
+
+// verifyEdgelessDB requests the /signature from the edb, it should match the hash of the manifest we expected
+func verifyEdgelessDB(edbHost string, m *manifest, edbPEM string) error {
+	b, err := json.Marshal(m)
+	if err != nil {
+		return fmt.Errorf("failed to marshal manifest to json - %w", err)
+	}
+	h := sha256.Sum256(b)
+	expectedHash := hex.EncodeToString(h[:])
+
+	url := fmt.Sprintf("https://%s:%s/signature", edbHost, edbHTTPPort)
+	req, err := http.NewRequestWithContext(context.Background(), "GET", url, nil)
+	if err != nil {
+		return fmt.Errorf("faild to create edb signature req - %w", err)
+	}
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM([]byte(edbPEM))
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs:    caCertPool,
+				MinVersion: tls.VersionTLS12,
+			},
+		},
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		return fmt.Errorf("/signature request failed - %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode < 200 || resp.StatusCode > 299 {
+		var msg []byte
+		_, err := resp.Body.Read(msg)
+		if err != nil {
+			return fmt.Errorf("edb /signature req failed with status code: %d, failed to read status text", resp.StatusCode)
+		}
+		return fmt.Errorf("edb /signature req failed with status: %d %s", resp.StatusCode, msg)
+	}
+
+	var edbHash []byte
+	edbHash, err = io.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("failed to read edbHash from /signature response - %w", err)
+	}
+	if expectedHash != string(edbHash) {
+		return fmt.Errorf("hash from edb /signature request didn't match expected hash of manifest.json, expected=%s, found=%s resp=%v", expectedHash, edbHash, resp)
+	}
+	log.Info("EDB signature matched the expected hash from our manifest (%s)", expectedHash)
+
+	return nil
+}
+
+// create Go standard database/sql connection to edb using a mysql driver
+func connectToEdgelessDB(edbHost string, edbPEM string) (*sql.DB, error) {
+	caCertPool := x509.NewCertPool()
```

Comment:
@BedrockSquirrel This seems to be the third time you've built this cert pool as well. Maybe build it once, and pass it around?